test/e2e/es/license_test.go

```diff
@@ -160,7 +159,6 @@ func TestEnterpriseTrialExtension(t *testing.T) {
 				},
 			},
 			licenseTestContext.DeleteAllEnterpriseLicenseSecrets(),
-			licenseTestContext.CreateEnterpriseTrialLicenseSecret(trialSecretName),
```

Comment:
isn't the main objective of this test to verify that an existing trial license can be extended with a non-trial short-lived license?
If we remove the trial license don't we remove half the purpose of this test?

Reply:
It was not testing this anyway. ECK-managed trials can only be started once. `TestEnterpriseTrialLicense` which runs just before this test, already starts an ECK-managed trial. So this statement was just a left-over piece of code an if you check the assertions it was never asserted that it had any effect (otherwise the test would fail)

What we could do instead to test that the trial extension is actually extending the trial is make both tests mutually exclusive:
* on production builds we just run the `TestEnterpriseTrialLicense` (would test only ECK-managed trial)
* on snapshot builds we run `TestEnterpriseTrialExtension` (would test both ECK-managed trial and extension)